test/unit/lib/tasks/asset_manager_test.rb

```diff
@@ -0,0 +1,168 @@
+    let(:edition) { create(:published_publication) }
+    let(:attachment) { create(:file_attachment_with_asset, attachable: edition) }
+
+    context "when asset is not deleted from asset manager" do
+      before do
+        attachment.attachment_data.assets.each do |asset|
+          stub_asset_manager_has_an_asset(asset.asset_manager_id, { deleted: false, file_url: asset.filename })
+        end
+      end
+
+      test "it should not include the attachment in the report" do
+        assert_output(/^(?!#{attachment.attachment_data.assets.first.filename},#{edition.public_url},#{edition.state}).*$/) { task.invoke }
+      end
+    end
+
+    context "when asset is deleted from asset manager" do
+      before do
+        attachment.attachment_data.assets.each do |asset|
+          stub_asset_manager_has_an_asset(asset.asset_manager_id, { deleted: true, file_url: asset.filename })
+        end
+      end
+
+      test "it should not include the attachment in the report" do
+        assert_output(/^(?!#{attachment.attachment_data.assets.first.filename},#{edition.public_url},#{edition.state}).*$/) { task.invoke }
+      end
+    end
+  end
+
+  context "for an asset attached only to both published and superseded editions" do
+    let(:attachment) { create(:file_attachment_with_asset) }
+    let(:edition_1) { create(:superseded_publication, :with_alternative_format_provider, attachments: [attachment]) }
+    let(:edition_2) { create(:published_publication, :with_alternative_format_provider, attachments: [attachment]) }
+
+    context "when asset is not deleted from asset manager" do
+      before do
+        attachment.attachment_data.assets.each do |asset|
+          stub_asset_manager_has_an_asset(asset.asset_manager_id, { deleted: false, file_url: asset.filename })
+        end
+      end
+
+      test "it should not include the attachment in the report" do
+        assert_output(/^(?!#{attachment.attachment_data.assets.first.filename},(#{Regexp.escape(edition_1.public_url)}|#{Regexp.escape(edition_2.public_url)}),(#{edition_1.state}|#{edition_2.state})).*$/) { task.invoke }
+      end
+    end
+
+    context "when asset is deleted from asset manager" do
+      before do
+        attachment.attachment_data.assets.each do |asset|
+          stub_asset_manager_has_an_asset(asset.asset_manager_id, { deleted: true, file_url: asset.filename })
+        end
+      end
+
+      test "it should not include the attachment in the report" do
+        assert_output(/^(?!#{attachment.attachment_data.assets.first.filename},(#{Regexp.escape(edition_1.public_url)}|#{Regexp.escape(edition_2.public_url)}),(#{edition_1.state}|#{edition_2.state})).*$/) { task.invoke }
+      end
+    end
+  end
+
+  context "for an asset attached only to something that is not an edition" do
+    let(:attachment) { create(:file_attachment_with_asset, attachable: create(:double)) }
+
+    test "it should include no output in the report" do
+      assert_output("") { task.invoke }
```

Comment:
Is there a reason you can't do this in other tests? Will/might they have some output depending on factory randomness?

Reply:
I was also considering adding a header row to the output, but never did, which is why I'm using a regex, rather than looking at the whole output.  There are some cases where something is output and we want to check the other isn't, so I've tried to keep them consistent.